### ISSUE TITLE
Implement GenParticle phase space generator with Jax

### ISSUE THREAD

**Issue by VitalyVorobyev** [VitalyVorobyev/decay-gen#5]
Currently, scikit-hep phasespace is used to produce phase space decays. It uses Tensroflow making HelixGen dependent on Tensorflow, too. Once phasespace generation is implemented with Jax, HelixGen will be dependent only on Jax

**Comment by jonas-eschle** [comment]
This sounds interesting, and in fact we can even implement this maybe upstreams. We had the same decision to make on what library(ies) to build on, and TensorFlow seemed the best option from many aspects. And then, as it works, it works.

So out of curiosity: Why would you want to change to Jax, is there an advantage? Is TensorFlow a too-heavy dependency?

**Comment by VitalyVorobyev** [comment]
The `phasespace` package works just fine and it is a great tool! The main motivation is avoiding cast of TF objects to Jax objects inside `helixgen`. A simpler solution is to switch to TensorFlow, but I like Jax. Implementing the `phasespace` algorithm with Jax is a good exercise to get used with Jax and learn generic algorithm for the phase space decay.

One possible advantage is that Jax supports both reverse-mode and forward-mode automatic differentiation while TensorFlow and PyTorch support only one of them (I don't remember which one). If used wise, it can speed-up computations for some data.

**Comment by jonas-eschle** [comment]
That sounds good and I think we would be quite interested to even have another backend for phasespace. So if you implement that, mixing it into the existing phasespace package (if it provides synergies such as using existing tests, docs etc) could be quite beneficial (but thinking in long term plans).

I am mostly wondered however on what is the best choice of backend, as it is a more general question for different libraries. The JAX vs PyTorch vs TensorFlow is an ongoing discussion, so good arguments for _either_ package are always interesting to hear.

About the diff-mode: By default they have reverse-mode, however TensorFlow also has forward-mode (https://www.tensorflow.org/api_docs/python/tf/autodiff/ForwardAccumulator) which we use for example in zfit for hessian computation (which, yes, is orders of magnitudes faster).
